src/components/WixUiTpaConfig/WixUiTpaConfig.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+export interface WixUiTpaConfig {
+  mobile: boolean;
+}
+
+export interface WixUiTpaConfigProps {
+  mobile?: boolean;
+}
+
+const wixUiTpaConfig = React.createContext<WixUiTpaConfig>({
+  mobile: false,
+});
+
+export const WixUiTpaConfigProvider = wixUiTpaConfig.Provider;
+
+export const WixUiTpaConfigConsumer = wixUiTpaConfig.Consumer;
+
+export const withWixUiTpaConfig = <TProps extends object>(
```

Comment:
I don't like the extra wrapper. I prefer components just to consume the context consumer

Reply:
@shlomitc the use of the Consumer Should be like `class Button extends React.Component<ButtonProps> {
  render() {
    const {
      priority,
      size,
      fullWidth,
      ...rest
    } = this.props;
    return (<WixUiTpaContextConsumer>
      {({mobile}) => <ButtonNext
        {...rest}
        {...style('root', { priority, size, fullWidth, mobile }, rest)}
      />}
    </WixUiTpaContextConsumer>)
  }
}`?